server/resource_server/service/service.go

```diff
@@ -2,22 +2,32 @@ package service
 
 import (
 	"fmt"
+	"github.com/go-playground/validator/v10"
 	"globe-and-citizen/layer8/server/resource_server/dto"
+	"globe-and-citizen/layer8/server/resource_server/emails/verification"
 	interfaces "globe-and-citizen/layer8/server/resource_server/interfaces"
 	"globe-and-citizen/layer8/server/resource_server/models"
 	"globe-and-citizen/layer8/server/resource_server/utils"
-
-	"github.com/go-playground/validator/v10"
+	"time"
 )
 
 type service struct {
-	repository interfaces.IRepository
+	repository    interfaces.IRepository
+	emailVerifier *verification.EmailVerifier
+
+	verificationCodeValidityDuration time.Duration
```

Comment:
My intuition tells me that `verificationCodeValidityDuration time.Duration` should somehow be captured in the `emailVerifier` implementation. 

```
type EmailVerifier struct {
    adminEmailAddress string

    emailSenderService sender.Service
    codeGenerator      code.Generator

    now func() time.Time
    verificationCodeValidityDuration time.Duration
}
```

It just fits so nicely inside verification.EmailVerifier no?

Reply:
Line 122 of `main.go` is the perfect opportunity to add the `verificationCodeValidityDuration`

```
	emailVerifier := verification.NewEmailVerifier(
		adminEmailAddress,
		sender.NewMailerSendService(
			os.Getenv("MAILER_SEND_API_KEY"),
			os.Getenv("MAILER_SEND_TEMPLATE_ID"),
		),
		code.NewRandomCodeGenerator(verificationCodeSize),
		time.Now,
                verificationCodeValidityDuration
	)
```

Reply:
Thanks for the feedback! Added ```verificationCodeValidityDuration``` as the field into the ```EmailVerifier```, as you suggested.